pkg/asset/installconfig/vsphere/client.go

```diff
@@ -2,30 +2,118 @@ package vsphere
-// mostly used to create a mock client that can be used for testing.
-type Finder interface {
-	Datacenter(ctx context.Context, path string) (*object.Datacenter, error)
-	DatacenterList(ctx context.Context, path string) ([]*object.Datacenter, error)
-	DatastoreList(ctx context.Context, path string) ([]*object.Datastore, error)
-	ClusterComputeResource(ctx context.Context, path string) (*object.ClusterComputeResource, error)
-	ClusterComputeResourceList(ctx context.Context, path string) ([]*object.ClusterComputeResource, error)
-	Folder(ctx context.Context, path string) (*object.Folder, error)
-	NetworkList(ctx context.Context, path string) ([]object.NetworkReference, error)
-	Network(ctx context.Context, path string) (object.NetworkReference, error)
-	ResourcePool(ctx context.Context, path string) (*object.ResourcePool, error)
+// CreateVSphereClients creates the SOAP and REST client to access
+// different portions of the vSphere API
+// e.g. tags are only available in REST
+func CreateVSphereClients(ctx context.Context, vcenter, username, password string) (*vim25.Client, *rest.Client, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+
+	u, err := soap.ParseURL(vcenter)
+	if err != nil {
+		return nil, nil, err
+	}
+	u.User = url.UserPassword(username, password)
+	c, err := govmomi.NewClient(ctx, u, false)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	restClient := rest.NewClient(c.Client)
+	err = restClient.Login(ctx, u.User)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return c.Client, restClient, nil
 }
 
-// NewFinder creates a new client that conforms with the Finder interface and returns a
-// vmware govmomi finder object that can be used to search for resources in vsphere.
-func NewFinder(client *vim25.Client, all ...bool) Finder {
-	return find.NewFinder(client, all...)
+// getNetworks returns a slice of Managed Object references for the given vSphere Cluster.
+func getNetworks(ctx context.Context, ccr *object.ClusterComputeResource) ([]types.ManagedObjectReference, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+	var ccrMo mo.ClusterComputeResource
+
+	err := ccr.Properties(ctx, ccr.Reference(), []string{"network"}, &ccrMo)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get properties of cluster")
+	}
+	return ccrMo.Network, nil
+}
+
+// GetClusterNetworks returns a slice of Managed Object references for vSphere networks in the given Datacenter
+// and Cluster. The given NetworkIdentifier and Finder are used to query vSphere API.
+func GetClusterNetworks(ctx context.Context, finder *find.Finder, datacenter, cluster string) ([]types.ManagedObjectReference, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+
+	// Get vSphere Cluster resource in the given Datacenter.
+	path := fmt.Sprintf("/%s/host/%s", datacenter, cluster)
```

Comment:
@jcpowermac I'm afraid if a customer uses a folder for their clusters this will break. Do you know if this is common and we should account for it?

Reply:
Yep

Reply:
I don't think this is a concern. We would have hit this already since the terraform provider does this:
https://github.com/openshift/installer/blob/master/terraform/providers/vsphereprivate/resource_vsphereprivate_import_ova.go#L148